### PR TITLE
fix: or conditions

### PR DESCRIPTION
**Pull request by chenshoo**

### CODE REVIEW

File: packages/polaris-typeorm/src/handlers/find-handler.ts [hunk 1]
```diff
@@ -1,50 +1,85 @@
...
+        criteria: any,
+        includeLinkedOper: boolean,
+        headers: PolarisRequestHeaders,
+        qb: SelectQueryBuilder<any>,
+    ) {
+        if ((criteria as any)?.where?.realityId === undefined) {
+            const parameters = {
+                realityId: realityIdCriteria(includeLinkedOper, headers),
+            };
+            const realityId = `${qb.alias}.realityId = :realityId`;
```
Comment (ronkatz96) [inline packages/polaris-typeorm/src/handlers/find-handler.ts:68]: `realityIdQuery` maybe?

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 2]
```diff
@@ -193,11 +194,16 @@ function applyDataVersionWhereConditions(
...
     }
     if (dataVersion > 0) {
-        qb.where(`${qb.alias}.dataVersion > :dataVersion`, { dataVersion });
-        names = names.slice(1);
-        for (const name of names) {
-            qb = qb.orWhere(`${name}.dataVersion > :dataVersion`);
-        }
+        qb = setWhereCondition(
+            qb,
+            new Brackets((qb2) => {
```
Comment (ronkatz96) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts]: maybe export to function?

File: packages/polaris-typeorm/src/handlers/find-handler.ts [hunk 3]
```diff
@@ -1,50 +1,85 @@
...
+            qb = setWhereInIdsCondition(qb, criteria);
         } else {
-            polarisCriteria = criteria || {};
+            if (criteria && criteria.where) {
+                if (criteria.where.deleted !== undefined) {
+                    this.deleteFindConditionIfRedundant(criteria.where);
+                    shouldAddDeleteCondition = false;
+                }
+                if (criteria.where instanceof Array && criteria.where.length > 0) {
+                    qb = setWhereCondition(
```
Comment (ronkatz96) [inline packages/polaris-typeorm/src/handlers/find-handler.ts]: maybe export to function?

File: packages/polaris-typeorm/test/unit-tests/handlers/find-handler.test.ts [hunk 4]
```diff
@@ -1,66 +1,136 @@
 import { PolarisGraphQLContext } from '@enigmatis/polaris-common';
-import { FindManyOptions, In } from 'typeorm';
+import { In } from 'typeorm';
 import { FindHandler } from '../../../src/handlers/find-handler';
-
+const createNewQB = () => {
```
Comment (osher-sade) [inline packages/polaris-typeorm/test/unit-tests/handlers/find-handler.test.ts]: rename to `createQueryBuilderMock`

File: packages/polaris-typeorm/src/typeorm-bypasses/polaris-entity-manager.ts [hunk 5]
```diff
@@ -198,6 +202,7 @@ export class PolarisEntityManager extends EntityManager {
             );
             const { ids, lastId } = this.getSortedIdsToReturnByPageSize(result);
             this.updateOnlinePaginatedContext(ids, result, lastId);
+            delete criteria?.where;
```
Comment (osher-sade) [inline packages/polaris-typeorm/src/typeorm-bypasses/polaris-entity-manager.ts]: why

**Review by osher-sade — APPROVED** [review]

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 6]
```diff
@@ -182,6 +184,20 @@ function joinDataVersionRelations(
     return qb;
 }
 
+function getOrDataVersionCondition(
+    qb: SelectQueryBuilder<any>,
+    dataVersion: number,
+    names: string[],
+) {
```
Comment (ronkatz96) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts]: add return type

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 7]
```diff
@@ -182,6 +184,20 @@ function joinDataVersionRelations(
     return qb;
 }
 
+function getOrDataVersionCondition(
+    qb: SelectQueryBuilder<any>,
```
Comment (osher-sade) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts]: rename to `queryBuilder`

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 8]
```diff
@@ -182,6 +184,20 @@ function joinDataVersionRelations(
...
+    dataVersion: number,
+    names: string[],
+) {
+    return new Brackets((qb2) => {
```
Comment (osher-sade) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts]: `qb2` => `qb`

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 9]
```diff
@@ -182,6 +184,20 @@ function joinDataVersionRelations(
...
+        qb2.where(`${qb.alias}.dataVersion > :dataVersion`, { dataVersion });
+        names = names.slice(1);
```
Comment (osher-sade) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts]: why?

File: packages/polaris-typeorm/src/handlers/data-version-handler.ts [hunk 10]
```diff
@@ -249,10 +267,12 @@ export const InnerJoinDataVersionQuery = (
 function buildInnerJoinQuery(
     selectQueries: SelectQueryBuilder<any>[],
     rootEntityIdSelection: string,
-): string {
+): { query: string; parameters: any } {
     let finalQuery = 'WITH w1(id, dv) AS (';
     const union = ' UNION ';
+    const parameters: any[] = [];
     selectQueries.forEach((query: SelectQueryBuilder<any>) => {
+        parameters.push(...selectQueries[0].getQueryAndParameters()[1]);
```
Comment (osher-sade) [inline packages/polaris-typeorm/src/handlers/data-version-handler.ts]: why?

File: packages/polaris-typeorm/src/handlers/find-handler.ts [hunk 11]
```diff
@@ -1,50 +1,90 @@
...
+            shouldAddDeleteCondition = false;
+        }
+        if (shouldAddDeleteCondition && !shouldIncludeDeletedEntities) {
+            qb = setWhereCondition(qb, `${qb.alias}.deleted = :deleted`, { deleted: false });
+        }
+        return qb;
+    }
 
-        let polarisCriteria: any;
+    public static applyUserConditions<Entity>(criteria: any, qb: SelectQueryBuilder<any>) {
```
Comment (osher-sade) [inline packages/polaris-typeorm/src/handlers/find-handler.ts]: rename to `applyCustomCriteria`

File: packages/polaris-typeorm/src/handlers/find-handler.ts [hunk 12]
```diff
@@ -1,50 +1,90 @@
...
-        polarisCriteria.where = { ...polarisCriteria.where };
-        if (polarisCriteria.where.deleted === undefined && !shouldIncludeDeletedEntities) {
-            polarisCriteria.where.deleted = false;
-        } else {
-            this.deleteFindConditionIfRedundant(polarisCriteria);
-        }
-        if (polarisCriteria.where.realityId === undefined) {
-            polarisCriteria.where.realityId = realityIdCriteria(includeLinkedOper, headers);
+    private static createOrWhereCondition(criteria: any) {
+        return new Brackets((qb2) => {
```
Comment (osher-sade) [inline packages/polaris-typeorm/src/handlers/find-handler.ts]: rename to `qb`

**Review by osher-sade — APPROVED** [review]

File: packages/polaris-typeorm/src/handlers/find-handler.ts [hunk 13]
```diff
@@ -1,50 +1,90 @@
...
+        FindHandler.applyUserConditions(criteria, qb);
+        this.applyDeleteCondition(criteria as any, shouldIncludeDeletedEntities, qb);
+        return qb;
+    }
+
+    private applyDeleteCondition<Entity>(
+        criteria: any,
+        shouldIncludeDeletedEntities: boolean,
+        qb: SelectQueryBuilder<any>,
+    ) {
```
Comment (ronkatz96) [inline packages/polaris-typeorm/src/handlers/find-handler.ts]: return type

**Review by ronkatz96 — APPROVED** [review]

**Pull request merged**